app/components/c-swiper-modal.vue

```diff
@@ -0,0 +1,290 @@
+<template>
+  <div :class="b()">
+    <div ref="gallery" :class="b('container swiper-container gallery-top')">
+      <div :class="b('wrapper swiper-wrapper')">
+        <div v-for="picture in pictures" :class="b('slide swiper-slide')" :key="picture.id">
+          <e-picture
+            :sizes="sizes"
+            :srcset="picture.srcset"
+            :fallback="picture.fallback"
+            :alt="picture.alt"/>
+        </div>
+      </div>
+      <div :class="b('button-prev swiper-button-prev')"></div>
+      <div :class="b('button-next swiper-button-next')"></div>
+    </div>
+    <div ref="thumbnails" :class="b('container swiper-container gallery-thumbs')">
+      <div :class="b('wrapper swiper-wrapper')">
+        <div v-for="picture in pictures" :class="b('slide swiper-slide')" :key="picture.id">
+          <e-picture
+            :sizes="sizes"
+            :srcset="picture.srcset"
+            :fallback="picture.fallback"
+            :alt="picture.alt"/>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+
+  import Swiper from 'swiper';
+  import { BREAKPOINTS } from '@/setup/globals';
+  import 'swiper/dist/css/swiper.css';
+
+  /**
+   * Touch enabled slider component with thumbnails based on
+   * [swiper](http://idangero.us/swiper/api/).
+   */
+  export default {
+    name: 'c-swiper-modal',
+    // components: {},
+
+    // mixins: [],
+
+    props: {
+      /**
+       * Gallery images passed to swiper
+       */
+      images: {
+        type: Array,
+        required: true,
+      },
+
+      /**
+       * Sets the initial slide index
+       */
+      initialSlide: {
+        type: Number,
+        default: 0,
+      },
+
+      /**
+       * Swiper options passed by component
+       */
+      options: {
+        type: Object,
+        default: () => {},
+      },
+    },
+
+    data() {
+      return {
+        optionsDefault: {
+          slidesPerGroup: 1,
+          watchOverflow: true,
+          keyboard: {
+            enabled: true,
+          },
+          lazy: false,
+          navigation: {
+            nextEl: '.swiper-button-next',
+            prevEl: '.swiper-button-prev',
+            hideOnClick: true,
+          },
+        },
+        optionsThumbnails: {
+          slidesPerGroup: 1,
+          watchOverflow: true,
+          lazy: false,
+          centeredSlides: true,
+          slidesPerView: 3,
+          slideToClickedSlide: true,
+          roundLengths: true,
+        },
+        swiperGallery: null,
+        swiperThumbnails: null,
+        sizes: BREAKPOINTS,
+        slidesPerBreakpoint: {
+          xxs: 3,
+          xs: 3,
+          sm: 5,
+          md: 7,
+          lg: 9,
+          xl: 9,
+        },
+        resizeTimerId: null,
+      };
+    },
+
+    computed: {
+      optionsMerged() {
+        return {
+          ...this.optionsDefault,
+          ...this.options,
+        };
+      },
+
+      pictures() {
```

Comment:
- [ ] Looks like the same function like in the c-swiper-gallery component. Can't you combine them? Or if both needed store in a helper "pictureMapping" function?